modules/nw-ingress-controller-nodeportservice-projects.adoc

```diff
@@ -0,0 +1,190 @@
++
+.Example of a CR file that defines information for the `IngressController` object
+[source,yaml]
+----
+apiVersion: v1
+items:
+- apiVersion: operator.openshift.io/v1
+  kind: IngressController
+  metadata:
+    name: <custom_name> <1>
+    namespace: openshift-ingress-operator
+  spec:
+    replicas: 1
+    domain: <domain> <2>
+    nodePlacement:
+      nodeSelector:
+        matchLabels:
+          <key>: <value> <3>
+    namespaceSelector:
+     matchLabels:
+       <key>: <value> <4>
+    endpointPublishingStrategy:
+      type: NodePortService 
+kind: List
+metadata:
+  resourceVersion: ""
+  selfLink: ""
+# ...
+----
+<1> Specify the a custom `name` for the `IngressController` CR.
+<2> The DNS name that the Ingress Controller services.
+<3> Specify the label for the nodes that will include the custom Ingress Controller.
+<4> Specify the label for a set of namespaces. Substitute `<key>:<value>` with a map of key-value pairs where `<key>` is a unique name for the new label and `<value>` is its value. For example: `ingresscontroller: custom-ic`.
+
+. Add a label to a node by using the `oc label node` command:
++
+[source,terminal]
+----
+$ oc label node <node_name> <key>=<value> <1>
+----
+<1> Where `<value>` must match the key-value pair that you specified in the `nodePlacement` section of your `IngressController` CR.
+
+. Create the `IngressController` object:
++
+[source,terminal]
+----
+$ oc create -f <ingress_controller_cr>.yaml
+----
+
+. Find the port for the service that was created for the `IngressController` CR:
++
+[source,terminal]
+----
+$ oc get svc -n openshift-ingress
+----
++
+.Example output that shows port `80:32432/TCP` for the `router-nodeport-custom-ic3` service
+[source,terminal]
+----
+NAME                        TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)                                     AGE
+router-internal-default     ClusterIP   172.30.195.74    <none>        80/TCP,443/TCP,1936/TCP                     223d
+router-nodeport-custom-ic3   NodePort    172.30.109.219   <none>        80:32432/TCP,443:31366/TCP,1936:30499/TCP   155m
+----
+
+. Use the `oc new-project` command to create a new project:
++
+[source,terminal]
+----
+$ oc new-project <project_name>
+----
+
+. Enter the following command to label the new namespace:
++
+[source,terminal]
+----
+$ oc label namespace <project_name> <key>=<value> <1>
+----
+<1> Where `<key>=<value>` must match the value that you specified in the `namespaceSelector` section of your Ingress Controller CR.
+
+. Create a new application in your cluster:
++
+[source,terminal]
+----
+$ oc new-app --image=<image_name> <1>
+----
+<1> An example of `<image_name>` is `quay.io/openshifttest/hello-openshift:multiarch`.
+
+. Create a `Route` object for a service, so that the pod can use the service to expose the application external to the cluster.
++
+[source,terminal]
+----
+$ oc expose svc/<service_name> --hostname=<svc_name>.<project_name>.<domain_name> <1>
```

Comment:
The hostname should match the format: `*.<domain_name>`, this <domain_name> should be the one we specified in custom ingresscontroller
```
  domain: <domain> <2>
```
So we should use dash `-` instead of dot `.` between <svc_name> and <project_name>, the correct one should be like `--hostname=<svc_name>-<project_name>.<domain_name>`
To make it more clear, maybe we could use the word `custom_ic_domain_name>` here and the definition.

Reply:
:heavy_check_mark: